src/io/write_ascii.cpp

```diff
@@ -91,14 +89,14 @@ namespace wash {
 
                 outputFile << sep << particle.get_smoothing_length();
 
-                for (auto& force : forces_vector) {
+                for (auto& force : get_force_vectors()) {
                     for (size_t i = 0; i < DIM; i++) {
-                        outputFile << sep << particle.get_force_vector(force).at(i);
+                        outputFile << sep << (*force)[particle.get_id()][i];
```

Comment:
If changed to references instead of pointers (from comment above) we can remove the manual dereference of `force`